ui/src/shared/components/cells/Cells.tsx

```diff
@@ -10,24 +12,37 @@ import GradientBorder from 'src/shared/components/cells/GradientBorder'
 
 // Utils
 import {fastMap} from 'src/utils/fast'
+import {getByID} from 'src/resources/selectors'
 
 // Constants
 import {LAYOUT_MARGIN, DASHBOARD_LAYOUT_ROW_HEIGHT} from 'src/shared/constants'
 
 // Types
-import {Cell, TimeRange, RemoteDataState} from 'src/types'
+import {
+  AppState,
+  Cell,
+  TimeRange,
+  RemoteDataState,
+  ResourceType,
+  View,
+} from 'src/types'
 
 import {ErrorHandling} from 'src/shared/decorators/errors'
 
+interface StateProps {
+  state: AppState
+}
+
 interface Props {
   cells: Cell[]
   timeRange: TimeRange
   manualRefresh: number
   onPositionChange?: (cells: Cell[]) => void
 }
+type AllProps = StateProps & Props
```

Comment:
if we're creating a custom type here, we should add `WithRouterProps` as an interface here:

`type AllProps = StateProps & Props & WithRouterProps` 

That way we can write the component as such:

```
class Cells extends Component<AllProps> {
...
```

Reply:
done